tests/e2e/cloudprovider/test-lb-service.sh

```diff
@@ -801,6 +801,160 @@ EOF
+	if [ "$FIP_IN_OS" -ne "1" ]; then
+	    printf "The floating IP doesn't exist in openstack"
+	    exit 1
+	fi
+
+	printf "\n>>>>>>> Delete Service ${service}\n"
+	kubectl -n $NAMESPACE delete service ${service}
+
+	printf "\n>>>>>>> Checking if Floating IP is still in openstack"
+	FIP_IN_OS=`openstack floating ip list --floating-ip-address "$FIP" -f json | jq length`
+	if [ "$FIP_IN_OS" -ne "1" ]; then
+	    printf "The floating IP doesn't exist in openstack"
+	    exit 1
+	fi
+
+		printf "\n>>>>>>> Creating new service with the same floating ip"
+	
+	cat <<EOF | kubectl apply -f -
+kind: Service
+apiVersion: v1
+metadata:
+  name: ${second_service}
+  namespace: $NAMESPACE
+  annotations:
+    loadbalancer.openstack.org/keep-floatingip: "true"
+    loadbalancer.openstack.org/load-balancer-address: "$FIP"
+spec:
+  type: LoadBalancer
+  selector:
+    run: echoserver
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 8080
+EOF
+	printf "\n>>>>>>> Waiting for the Service ${second_service} creation finished\n"
+	wait_for_service_address ${second_service}
+	
+	printf "\n>>>>>>> Delete Service ${second_service}\n"
+	kubectl -n $NAMESPACE delete service ${second_service}
+
+	printf "\n>>>>>>> Delete floating IP ${FIP_IN_OS}\n"
+	FIP_ID=`openstack floating ip list --floating-ip-address "$FIP" -f json | jq '.[0].ID' | tr -d '"'`
+	openstack floating ip delete $FIP_ID
+}
+########################################################################
+## Name: test_healthmonitors
+## Desc: The steps in this test case:
+##   1. Create load balanccer with healthmonitor with non-default values
+##   2. Using the openstack CLI check if:
+##    2.1 The load balancer has health monitor
+##    2.2 The health monitor matches the specification provided in the service via annotation
+##   3. Delete the service
+########################################################################
+function test_healthmonitors {
+	local service="test-healthmonitor"
+
+
+	printf "\n>>>>>>> Test - Creating healthmonitor\n"
+	printf ">>>>>>> Create Service ${service}\n"
+	cat <<EOF | kubectl apply -f -
+kind: Service
+apiVersion: v1
+metadata:
+  name: ${service}
+  namespace: $NAMESPACE
+  annotations:
+    loadbalancer.openstack.org/enable-health-monitor: "true"
+    loadbalancer.openstack.org/health-monitor-delay: "61"
+    loadbalancer.openstack.org/health-monitor-timeout: "32"
+    loadbalancer.openstack.org/health-monitor-max-retries: "7"
+    loadbalancer.openstack.org/health-monitor-max-retries-down: "2"
+spec:
+  type: LoadBalancer
+  selector:
+    run: echoserver
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 8080
+EOF
+
+	printf "\n>>>>>>> Waiting for the Service ${service} creation finished\n"
+	wait_for_service_address ${service}
+	printf "\n>>>>>>> Checking if Octavia health monitor is matching spec\n"
+	
+	HM_ID=`openstack loadbalancer healthmonitor list | grep "$service" | awk '{print $2}'`
```

Comment:
You could do `show` with an exact expected name. That could be useful if we'll add new tests with healthmonitors later.